src/test/test_chat_message_history.py

```diff
@@ -0,0 +1,180 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import random
+import re
+import string
+import time
+import uuid
+from multiprocessing import Process
+from typing import Iterator
+
+import pytest
+from google.cloud import bigtable
+from langchain_core.messages import AIMessage, BaseMessage, HumanMessage
+
+from langchain_google_bigtable.chat_message_history import BigtableChatMessageHistory
+
+TABLE_ID_PREFIX = "test-table-"
+
+
+@pytest.fixture
+def client() -> Iterator[bigtable.Client]:
+    yield bigtable.Client(
+        project=get_env_var("PROJECT_ID", "ID of the GCP project"), admin=True
+    )
+
+
+@pytest.fixture
+def instance_id() -> Iterator[str]:
+    yield get_env_var("INSTANCE_ID", "ID of the Cloud Bigtable instance")
+
+
+@pytest.fixture
+def table_id(instance_id: str, client: bigtable.Client) -> Iterator[str]:
+    table_id = TABLE_ID_PREFIX + "".join(
+        random.choice(string.ascii_lowercase) for _ in range(10)
+    )
+    # Create table
+    client.instance(instance_id).table(table_id).create()
+
+    yield table_id
+
+    # Teardown
+    client.instance(instance_id).table(table_id).delete()
+
+
+def test_bigtable_full_workflow(
+    instance_id: str, table_id: str, client: bigtable.Client
+) -> None:
+    session_id = uuid.uuid4().hex
+    history = BigtableChatMessageHistory(
+        instance_id, table_id, session_id, client=client
+    )
+
+    history.init_schema()
+    history.add_ai_message("Hey! I am AI!")
+    history.add_user_message("Hey! I am human!")
+    messages = history.messages
+
+    assert len(messages) == 2
+    assert isinstance(messages[0], AIMessage)
+    assert messages[0].content == "Hey! I am AI!"
+    assert isinstance(messages[1], HumanMessage)
+    assert messages[1].content == "Hey! I am human!"
+
+    history.clear()
+    assert len(history.messages) == 0
+
+
+def get_index_from_message(message: BaseMessage) -> int:
+    match = re.search("^Hey! I am (AI|human)! Index: ([0-9]+)$", str(message.content))
+    if match:
+        return int(match[2])
+    return 0
+
+
+def test_bigtable_loads_of_messages(
+    instance_id: str, table_id: str, client: bigtable.Client
+) -> None:
+    NUM_MESSAGES = 10000
+    session_id = uuid.uuid4().hex
+    history = BigtableChatMessageHistory(
+        instance_id, table_id, session_id, client=client
+    )
+
+    history.init_schema()
+
+    proc = []
+    for i in range(NUM_MESSAGES):
+        p = Process(
+            target=lambda i: history.add_ai_message(f"Hey! I am AI! Index: {2*i}"),
+            args=[i],
+        )
+        p.start()
+        proc.append(p)
+        p = Process(
+            target=lambda i: history.add_user_message(
+                f"Hey! I am human! Index: {2*i+1}"
+            ),
+            args=[i],
+        )
+        p.start()
+        proc.append(p)
```

Comment:
nit:  I question if we are really gaining anything here by using processes because of the GIL.

Reply:
Yes. I tried without it and the test takes forever =\
I am not a python expert, so I can't tell why it's behaving that way, but it does work much faster this way (takes 3~ minutes for the whole test, rather than over 20 before I gave up and stopped it)